internal/rbd/controllerserver.go

```diff
@@ -444,8 +444,17 @@ func (cs *ControllerServer) createVolumeFromSnapshot(ctx context.Context, cr *ut
 
 	// update parent name(rbd image name in snapshot)
 	rbdSnap.RbdImageName = rbdSnap.RbdSnapName
+	parentVol := generateVolFromSnap(rbdSnap)
+	// as we are operating on single cluster reuse the connection
+	parentVol.conn = rbdVol.conn
+	err = parentVol.openIoctx()
```

Comment:
Opening the ioctx is something internal to the rbdVolume struct, it should ideally only be done from `rbdVolume` methods. If it is needed to open the ioctx, do so in `cloneRbdImageFromSnapshot()` or whatever function requires it

Reply:
Thanks for the suggestion! Made the required changes.